File: python/easy/0389.countSegments.py
class Solution:
    def countSegments(self, s: str) -> int:
        return len(s.split())


if __name__ == "__main__":
    solution = Solution()
    print(solution.countSegments("Hello, my name is John"))  # 5
    print(solution.countSegments(""))  # 0
    print(solution.countSegments("" ""))  # 0
    print(
        solution.countSegments(
            "Of all the gin joints in all the towns in all the world,   "
        )
    )  # 13

File: javascript/easy/0389.countSegments.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function (s) {
  return s.trim() === "" ? 0 : s.split(/\s+/).filter((seg) => seg).length;
};

console.log(countSegments("Hello, my name is John")); // 5
console.log(countSegments("Hello")); // 1
console.log(countSegments("")); // 0
console.log(countSegments("                ")); // 0
console.log(countSegments("Of all the gin joints in all the towns in all the world,   ")); // 13
